v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/WriteToElasticsearch.java

```diff
@@ -66,14 +63,19 @@ public static Builder newBuilder() {
 
   public abstract ElasticsearchWriteOptions options();
 
+  private static final String DOCUMENT_TYPE="_doc";
+
   @Override
   public PDone expand(PCollection<String> jsonStrings) {
+    ConnectionInformation connectionInformation = new ConnectionInformation(options().getConnectionUrl());
 
     ElasticsearchIO.ConnectionConfiguration config =
         ElasticsearchIO.ConnectionConfiguration.create(
-            options().getTargetNodeAddresses().split(","),
+            new String[]{connectionInformation.getElasticsearchURL().toString()},
             options().getWriteIndex(),
-            options().getWriteDocumentType());
+            DOCUMENT_TYPE)
+            .withUsername(options().getWriteElasticsearchUsername())
```

Comment:
this should be `options().getElasticsearchUsername()` and `options().getElasticsearchPassword()`